src/main/java/mediator/ConcreteMediator.java

```diff
@@ -0,0 +1,287 @@
+     *
+     * @return A list of registered RUs.
+     */
+    @Override
+    public List<ManagedRadioUnit> getRegisteredRadioUnits() {
+        return radioUnits;
+    }
+
+    /**
+     * Prints a formatted list of RUs currently registered with the mediator.
+     */
+    @Override
+    public void printRegisteredRadioUnits() {
+        if (radioUnits.size() == 0) {
+            System.out.println("[ERROR] No RUs have been registered with the system.");
+        } else {
+            radioUnits.forEach(ru -> System.out.println(ru.getRadioUnitName()));
+        }
+    }
+
+    /**
+     * Print the RAT type for a RU associated with the specified name.
+     *
+     * @param name The name associated with the targeted RU.
+     */
+    @Override
+    public void printRatType(String name) {
+        AtomicBoolean flag = new AtomicBoolean(false);
+        radioUnits.forEach(ru -> {
+            if (ru.getRadioUnitName().equals(name)) {
+                flag.set(true);
+                System.out.println(ru.getRatType());
+                return;
+            }
+        });
+        if (!flag.get()) {
+            System.out.printf("[ERROR] No RUs exist with the name: %s%n", name);
+        }
+    }
+
+    /**
+     * Print the vendor for a RU associated with the specified name.
+     *
+     * @param name The name associated with the targeted RU.
+     */
+    @Override
+    public void printVendor(String name) {
+        AtomicBoolean flag = new AtomicBoolean(false);
+        radioUnits.forEach(ru -> {
+            if (ru.getRadioUnitName().equals(name)) {
+                flag.set(true);
+                System.out.println(ru.getVendor());
+                return;
+            }
+        });
+        if (!flag.get()) {
+            System.out.printf("[ERROR] No RUs exist with the name: %s%n", name);
+        }
+    }
+
+    /**
+     * Print the alarm for a RU associated with the specified name.
+     *
+     * @param name The name associated with the targeted RU.
+     */
+    @Override
+    public void printAlarmStatus(String name) {
+        AtomicBoolean flag = new AtomicBoolean(false);
+        radioUnits.forEach(ru -> {
+            if (ru.getRadioUnitName().equals(name)) {
+                flag.set(true);
+                System.out.println(ru.getAlarmStatus());
+                return;
+            }
+        });
+        if (!flag.get()) {
+            System.out.printf("[ERROR] No RUs exist with the name: %s%n", name);
+        }
+    }
+
+    /**
+     * Create a bare bones RU. In the future this method will handle the
+     * control logic associated with which RU is to be created. For now,
+     * We just create a generic RU with the passed parameters.
+     *
+     * @param name    The name of the RU.
+     * @param vendor  The vendor for the RU.
+     * @param ratType The RAT type for the RU.
+     */
+    @Override
+    public synchronized void createRu(String name, Vendor vendor, RatType ratType) {

```

Comment:
Indeed, although for the first demo we had decided to omit it since it technically wasn't a part of the requirement spec. Realistically instead of specifying an RU name in the UI, we should be specifying an IP (since it is valid for two RUs to share the same name according to the requirements). But still needs to be addressed for demo 2.